scripts/postcss/tests/css-url-to-variables.test.js

```diff
@@ -46,6 +47,21 @@ module.exports.tests = function tests() {
                 background: no-repeat url("./img/foo/bar/image.png");
             }`;
             await run(inputCSS, inputCSS, {}, assert);
+        },
+        "map is populated with icons": async (assert) => {
+            const compiledVariables = new Map();
+            const inputCSS = `div {
+                background: no-repeat center/80% url("../img/image.svg?primary=main-color--darker-20");
+            }
+            button {
+                background: url("/home/foo/bar/cool.svg?primary=blue&secondary=green");
+            }`;
+            const expectedObject = {
+                "icon-url-0": "../img/image.svg?primary=main-color--darker-20",
+                "icon-url-1": "/home/foo/bar/cool.svg?primary=blue&secondary=green",
+            };
+            await postcss([plugin({compiledVariables})]).process(inputCSS, { from: "/foo/bar/test.css", });
+            assert.deepEqual(expectedObject, compiledVariables.get("/foo/bar")["icon"]);
```

Comment:
are the compiled variables cached by the directory of the source file (rather than the source file name)? :thinking:

Reply:
Yep, so we'd have a directory like `/themes` and each theme would be in their own sub-directory, like `/themes/awesome-dark`, `/themes/hydrogen-default` ...
These theme directories are passed to the rollup plugin so that it knows where to find the `.json` and `.css`.
The source file name would always be something like `index.css` or `theme.css`.